src/main/java/wolox/training/models/User.java

```diff
@@ -59,23 +60,26 @@ public String getUsername() {
     }
 
     public void setUsername(String username) {
-        username = username;
+        Preconditions.checkNotNull(username);
+        this.username = username;
     }
 
     public String getName() {
         return name;
     }
 
     public void setName(String name) {
-        name = name;
+        Preconditions.checkNotNull(name);
+        this.name = name;
     }
 
     public LocalDate getBirthdate() {
         return birthdate;
     }
 
     public void setBirthdate(LocalDate birthdate) {
-        birthdate = birthdate;
+        Preconditions.checkNotNull(birthdate);
```

Comment:
After checking if a field is null, Add a check that the date of birth is before today's date.
An error message can be added to each check :

```
Preconditions.checkNotNull(birthdate, "The name birthdate cannot be null.");
Preconditions.checkArgument(!birthdate.isAfter(LocalDate.now()), "message");
this. birthdate = birthdate;
```

I recommend using a class to store error messages as constants.

Reply:
I put the messages and added it like constants.